pkg/api/types.go

```diff
@@ -16,6 +16,17 @@ type TypeMeta struct {
 	APIVersion string `json:"apiVersion,omitempty" yaml:"apiVersion,omitempty"`
 }
 
+// PullThroughRegistry contains configuration for pull-through registries
+type PullThroughRegistry struct {
```

Comment:
PullThroughRegistry doesn't seem like the right name for this, because it's not really doing any storage or caching. Maybe call it RegistryAuth?